server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,331 @@
+    try:
+        archive_tar_hex_value = md5sum(tb)
+    except Exception:
+        logger.exception("Quarantine: Could not read {}", tb)
+
+    return (archive_md5_hex_value, archive_tar_hex_value)
+
+
+def process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors, ef):
+
+    # Check for results that are ready for processing: version 002 agents
+    # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+    # after they are done with MD5 checking so that's what we look for.
+    list_check = glob.glob(
+        os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
+    )
+
+    archive = Path(config.ARCHIVE)
+    logger.info("{}", config.TS)
+    list_check.sort()
+    nstatus = ""
+
+    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+
+    for tbmd5 in list_check:
+        ntotal += 1
+
+        # full pathname of tarball
+        tb = Path(tbmd5[0:-4])
+        tbmd5 = Path(tbmd5)
+
+        # directory
+        tbdir = tb.parent
+
+        # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+        resultname = tb.name
+
+        controller = tbdir.name
+        dest = archive / controller
+
+        if (
+            (dest / resultname).is_file()
+            and (dest / tbmd5.name).is_file()
+            and (dest / resultname).exists()
+            and (dest / tbmd5.name).exists()
+        ):
+            ef.write(f"{config.TS}: Duplicate: {tb} duplicate name\n")
+            quarantine((duplicates / controller), logger, tb, tbmd5)
+            ndups += 1
+            continue
+
+        archive_tar_hex_value, archive_md5_hex_value = md5_check(tb, tbmd5, logger)
+        if archive_tar_hex_value != archive_md5_hex_value:
+            ef.write(f"{config.TS}: Quarantined: {tb} failed MD5 check\n")
+            logger.info("{}: FAILED", tb.name)
+            logger.info("md5sum: WARNING: 1 computed checksum did NOT match")
+            quarantine((qdir_md5 / controller), logger, tb, tbmd5)
+            nquarantined += 1
+            continue
+
+        # make the destination directory and its TODO subdir if necessary.
+        try:
+            os.makedirs(dest / "TODO")
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.error("{}: Error in creating TODO directory.", config.TS)
+            quarantine(os.path.join(errors, controller), logger, tb, tbmd5)
+            nerrs += 1
+            continue
+
+        # First, copy the small .md5 file to the destination. That way, if
+        # that operation fails it will fail quickly since the file is small.
+        try:
+            shutil.copy2(tbmd5, dest)
+        except Exception:
+            logger.error(
+                "{}: Error in copying .md5 file to Destination path.", config.TS
+            )
+            try:
+                os.remove(dest / tbmd5.name)
+            except Exception:
+                logger.error(
```

Comment:
If the copy failed, then it is possible the destination file does not exist.  So this error should only be logged if the exception is not "file not found".

Reply:
Seems fair